decoder/expr_literal_type_semtok.go

```diff
@@ -0,0 +1,153 @@
+package decoder
+
+import (
+	"context"
+
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func (lt LiteralType) SemanticTokens(ctx context.Context) []lang.SemanticToken {
+	typ := lt.cons.Type
```

Comment:
Is there a case where cons can be nil? Just to guard against a panic here

Reply:
It's a good point, although that would indicate invalid schema. So I think the best place to cover that would be in the schema validation. I will add some checks there!